--- linit.c.orig
+++ linit.c
@@ -50,6 +50,10 @@
 #include "third_party/lua/lualib.h"
 __static_yoink("lua_notice");

+extern int LuaRe(lua_State *L);
+extern int LuaPath(lua_State *L);
+extern int luaopen_argon2(lua_State *L);
+extern int luaopen_lsqlite3(lua_State *L);

 /*
 ** these libs are loaded by lua.c and are readily available to any Lua
@@ -66,6 +70,10 @@
   {LUA_UTF8LIBNAME, luaopen_utf8},
   {LUA_DBLIBNAME, luaopen_debug},
+  {"re", LuaRe},
+  {"path", LuaPath},
+  {"argon2", luaopen_argon2},
+  {"lsqlite3", luaopen_lsqlite3},
   {NULL, NULL}
 };

