--- BUILD.mk.orig	2025-01-05 22:05:49.000000000 +0000
+++ BUILD.mk	2025-12-15 00:16:08.477374726 +0000
@@ -83,6 +83,10 @@
 	third_party/lua/liolib.c				\
 	third_party/lua/llex.c					\
 	third_party/lua/llock.c					\
+	third_party/lua/lpath.c					\
+	third_party/lua/lre.c					\
+	third_party/lua/largon2.c				\
+	third_party/lua/lsqlite3.c				\
 	third_party/lua/lmathlib.c				\
 	third_party/lua/lmem.c					\
 	third_party/lua/lnotice.c				\
@@ -116,6 +120,7 @@
 	third_party/lua/lzio.c					\
 	third_party/lua/serialize.c				\
 	third_party/lua/visitor.c
+# note: lunix.c already compiled separately in lunix.a (see THIRD_PARTY_LUA_UNIX_A)

 THIRD_PARTY_LUA_A_OBJS =					\
 	$(THIRD_PARTY_LUA_A_SRCS:%.c=o/$(MODE)/%.o)
@@ -141,7 +146,10 @@
 	THIRD_PARTY_GDTOA					\
 	THIRD_PARTY_LINENOISE					\
 	THIRD_PARTY_MUSL					\
+	THIRD_PARTY_REGEX					\
 	THIRD_PARTY_TZ						\
+	THIRD_PARTY_ARGON2					\
+	THIRD_PARTY_SQLITE3

 THIRD_PARTY_LUA_A_DEPS :=					\
 	$(call uniq,$(foreach x,$(THIRD_PARTY_LUA_A_DIRECTDEPS),$($(x))))
