--- a/third_party/lua/BUILD.mk
+++ b/third_party/lua/BUILD.mk
@@ -224,6 +224,47 @@
 		$(foreach x,$(THIRD_PARTY_LUA_UNIX_DIRECTDEPS),$($(x)_A).pkg)
 
 ################################################################################
+# lpath.a
+
+THIRD_PARTY_LUA_PATH =						\
+	$(THIRD_PARTY_LUA_A_DEPS)				\
+	$(THIRD_PARTY_LUA_A)
+
+THIRD_PARTY_LUA_ARTIFACTS +=					\
+	THIRD_PARTY_LUA_PATH
+
+THIRD_PARTY_LUA_PATH_A =					\
+	o/$(MODE)/third_party/lua/lpath.a
+
+THIRD_PARTY_LUA_PATH_HDRS =					\
+	third_party/lua/lpath.h
+
+THIRD_PARTY_LUA_PATH_SRCS =					\
+	third_party/lua/lpath.c
+
+THIRD_PARTY_LUA_PATH_OBJS =					\
+	$(THIRD_PARTY_LUA_PATH_SRCS:%.c=o/$(MODE)/%.o)
+
+THIRD_PARTY_LUA_PATH_DIRECTDEPS =				\
+	LIBC_CALLS						\
+	LIBC_INTRIN						\
+	LIBC_STR						\
+	LIBC_SYSV						\
+	THIRD_PARTY_LUA
+
+THIRD_PARTY_LUA_PATH_DEPS :=					\
+	$(call uniq,$(foreach x,$(THIRD_PARTY_LUA_PATH_DIRECTDEPS),$($(x))))
+
+$(THIRD_PARTY_LUA_A):						\
+		third_party/lua/				\
+		$(THIRD_PARTY_LUA_PATH_A).pkg			\
+		$(THIRD_PARTY_LUA_PATH_OBJS)
+
+$(THIRD_PARTY_LUA_PATH_A).pkg:					\
+		$(THIRD_PARTY_LUA_PATH_OBJS)			\
+		$(foreach x,$(THIRD_PARTY_LUA_PATH_DIRECTDEPS),$($(x)_A).pkg)
+
+################################################################################
 # lua
 
 THIRD_PARTY_LUA_LUA_DIRECTDEPS =				\
@@ -240,6 +281,7 @@
 	THIRD_PARTY_LINENOISE					\
 	THIRD_PARTY_LUA						\
 	THIRD_PARTY_LUA_UNIX					\
+	THIRD_PARTY_LUA_PATH					\
 	TOOL_ARGS
 
 THIRD_PARTY_LUA_LUA_DEPS :=					\
