--- a/third_party/lua/BUILD.mk
+++ b/third_party/lua/BUILD.mk
@@ -14,6 +14,8 @@
 THIRD_PARTY_LUA_CHECKS =					\
 	$(THIRD_PARTY_LUA_A).pkg				\
 	$(THIRD_PARTY_LUA_UNIX).pkg				\
+	THIRD_PARTY_LUA_PATH					\
+	THIRD_PARTY_LUA_RE					\
 	$(THIRD_PARTY_LUA_HDRS:%=o/$(MODE)/%.ok)		\
 	o/$(MODE)/third_party/lua/lua.pkg			\
 	o/$(MODE)/third_party/lua/luac.pkg
@@ -224,6 +226,89 @@
 		$(foreach x,$(THIRD_PARTY_LUA_UNIX_DIRECTDEPS),$($(x)_A).pkg)
 
 ################################################################################
+# lpath.a
+
+THIRD_PARTY_LUA_PATH =						\
+	$(THIRD_PARTY_LUA_A_DEPS)				\
+	$(THIRD_PARTY_LUA_A)
+
+THIRD_PARTY_LUA_ARTIFACTS +=					\
+	THIRD_PARTY_LUA_PATH
+
+THIRD_PARTY_LUA_PATH_A =					\
+	o/$(MODE)/third_party/lua/lpath.a
+
+THIRD_PARTY_LUA_PATH_HDRS =					\
+	third_party/lua/lpath.h
+
+THIRD_PARTY_LUA_PATH_SRCS =					\
+	third_party/lua/lpath.c
+
+THIRD_PARTY_LUA_PATH_OBJS =					\
+	$(THIRD_PARTY_LUA_PATH_SRCS:%.c=o/$(MODE)/%.o)
+
+THIRD_PARTY_LUA_PATH_DIRECTDEPS =				\
+	LIBC_CALLS						\
+	LIBC_INTRIN						\
+	LIBC_STR						\
+	LIBC_SYSV						\
+	THIRD_PARTY_LUA
+
+THIRD_PARTY_LUA_PATH_DEPS :=					\
+	$(call uniq,$(foreach x,$(THIRD_PARTY_LUA_PATH_DIRECTDEPS),$($(x))))
+
+$(THIRD_PARTY_LUA_A):						\
+		third_party/lua/				\
+		$(THIRD_PARTY_LUA_PATH_A).pkg			\
+		$(THIRD_PARTY_LUA_PATH_OBJS)
+
+$(THIRD_PARTY_LUA_PATH_A).pkg:					\
+		$(THIRD_PARTY_LUA_PATH_OBJS)			\
+		$(foreach x,$(THIRD_PARTY_LUA_PATH_DIRECTDEPS),$($(x)_A).pkg)
+
+################################################################################
+# lre.a
+
+THIRD_PARTY_LUA_RE =						\
+	$(THIRD_PARTY_LUA_A_DEPS)				\
+	$(THIRD_PARTY_LUA_A)
+
+THIRD_PARTY_LUA_ARTIFACTS +=					\
+	THIRD_PARTY_LUA_RE
+
+THIRD_PARTY_LUA_RE_A =						\
+	o/$(MODE)/third_party/lua/lre.a
+
+THIRD_PARTY_LUA_RE_HDRS =					\
+	third_party/lua/lre.h
+
+THIRD_PARTY_LUA_RE_SRCS =					\
+	third_party/lua/lre.c
+
+THIRD_PARTY_LUA_RE_OBJS =					\
+	$(THIRD_PARTY_LUA_RE_SRCS:%.c=o/$(MODE)/%.o)
+
+THIRD_PARTY_LUA_RE_DIRECTDEPS =					\
+	LIBC_CALLS						\
+	LIBC_INTRIN						\
+	LIBC_STR						\
+	LIBC_SYSV						\
+	THIRD_PARTY_LUA						\
+	THIRD_PARTY_REGEX
+
+THIRD_PARTY_LUA_RE_DEPS :=					\
+	$(call uniq,$(foreach x,$(THIRD_PARTY_LUA_RE_DIRECTDEPS),$($(x))))
+
+$(THIRD_PARTY_LUA_A):						\
+		third_party/lua/				\
+		$(THIRD_PARTY_LUA_RE_A).pkg			\
+		$(THIRD_PARTY_LUA_RE_OBJS)
+
+$(THIRD_PARTY_LUA_RE_A).pkg:					\
+		$(THIRD_PARTY_LUA_RE_OBJS)			\
+		$(foreach x,$(THIRD_PARTY_LUA_RE_DIRECTDEPS),$($(x)_A).pkg)
+
+################################################################################
 # lua
 
 THIRD_PARTY_LUA_LUA_DIRECTDEPS =				\
